lib/SILOptimizer/Differentiation/PullbackCloner.cpp

```diff
@@ -164,7 +164,18 @@ class PullbackCloner::Implementation final
     assert(pbTupleTyple->getNumElements() == values.size() &&
            "The number of pullback tuple fields must equal the number of "
            "pullback struct element values");
-    auto res = pullbackTupleElements.insert({origBB, values});
+    auto res = pullbackTupleElements.insert({origBB, { values.begin(), values.end() }});
+    (void)res;
+    assert(res.second && "A pullback struct element already exists!");
```

Comment:
```suggestion
    assert(res.second && "A pullback tuple element already exists!");
```
(same below – there seem to be other "pullback struct" occurrences, if you want to update those too)